tests/v1/test_oracle.py

```diff
@@ -0,0 +1,135 @@
+# SPDX-License-Identifier: Apache-2.0
+import os
+
+import pytest
+
+from vllm import LLM
+from vllm.engine.arg_utils import AsyncEngineArgs
+
+UNSUPPORTED_MODELS_V1 = [
+    "openai/whisper-large-v3",  # transcription
+    "facebook/bart-large-cnn",  # encoder decoder
+    "mistralai/Mamba-Codestral-7B-v0.1",  # mamba
+    "ibm-ai-platform/Bamba-9B",  # hybrid
+    "BAAI/bge-m3",  # embedding
+]
+
+MODEL = "meta-llama/Llama-3.2-1B-Instruct"
+
+
+@pytest.mark.parametrize("model", UNSUPPORTED_MODELS_V1)
+def test_reject_unsupported_models(monkeypatch, model):
+    with monkeypatch.context() as m:
+        m.setenv("VLLM_USE_V1", "1")
+        args = AsyncEngineArgs(model=model)
+
+        with pytest.raises(NotImplementedError):
+            _ = args.create_engine_config()
+
+
+def test_unsupported_configs(monkeypatch):
+
+    with monkeypatch.context() as m:
+        m.setenv("VLLM_USE_V1", "1")
+
+        with pytest.raises(NotImplementedError):
+            AsyncEngineArgs(
+                model=MODEL,
+                kv_cache_dtype="fp8",
+            ).create_engine_config()
+
+        with pytest.raises(NotImplementedError):
+            AsyncEngineArgs(
+                model=MODEL,
+                speculative_model=MODEL,
+            ).create_engine_config()
+
+        with pytest.raises(NotImplementedError):
+            AsyncEngineArgs(
+                model=MODEL,
+                guided_decoding_backend="lm-format-enforcer:no-fallback",
+            ).create_engine_config()
+
+        with pytest.raises(NotImplementedError):
+            AsyncEngineArgs(
+                model=MODEL,
+                preemption_mode="swap",
+            ).create_engine_config()
+
+        with pytest.raises(NotImplementedError):
+            AsyncEngineArgs(
+                model=MODEL,
+                disable_async_output_proc=True,
+            ).create_engine_config()
+
+        with pytest.raises(NotImplementedError):
+            AsyncEngineArgs(
+                model=MODEL,
+                scheduling_policy="priority",
+            ).create_engine_config()
+
+        with pytest.raises(NotImplementedError):
+            AsyncEngineArgs(
+                model=MODEL,
+                num_scheduler_steps=5,
+            ).create_engine_config()
+
+        with pytest.raises(NotImplementedError):
+            AsyncEngineArgs(
+                model=MODEL,
+                scheduler_delay_factor=1.2,
+            ).create_engine_config()
+
+
+def test_enable_by_default_fallback(monkeypatch):
```

Comment:
We should consider the case where a user explicitly asks for an attention backend that isn't supported in V1 i.e. `VLLM_ATTENTION_BACKEND=XFORMERS`. It would be nice if we could fallback to V0 in this case

Reply:
good one

Reply:
added!